### DISCUSSION TITLE
How to launch RSS reader app from Safari because the xml pages looks ugly at mobile safari

### DISCUSSION THREAD

**Discussion by iamgiannisx** [razonyang/hugo-theme-bootstrap#1092] [Ideas]
I am trying to find a way to open xml links from mobile safari with reader app 

The solution is here --> https://forums.developer.apple.com/forums/thread/46790

Any help for this.

Thanks.

**Comment by razonyang** [comment]
Could you please provide a step-by-step reproduction and the expected result?

**Comment by iamgiannisx** [comment]
Reproduction 

https://share.icloud.com/photos/0f2NQm1xk8CfkhvP12Ql5a55w


Desired result 
https://share.icloud.com/photos/01chXxPTAmqVZbB8tpQbc-qpg


The solution is to add feed: in front of the url link, check the link at 4.89sec at the second video 


I try to add it with javascript but with no luck.


`function modifyFeedLinks() {
  const isSafari = /^((?!chrome|android).)*safari/i.test(navigator.userAgent);
  const isMobileSafari = /iPhone|iPod|iPad/.test(navigator.userAgent) 
      && /AppleWebKit/.test(navigator.userAgent) 
      && !/(Chrome|CriOS|OPiOS)/.test(navigator.userAgent);
  if (isSafari || isMobileSafari) {
      // Get all links
      const links = document.getElementsByTagName('a');
      // Loop through each link
      Array.from(links).forEach(link => {
          const href = link.getAttribute('href');
          // Check if href exists and doesn't already start with feed:
          if (href && !href.startsWith('feed:') && (href.endsWith('.xml') || href.endsWith('.rss'))) {
            link.setAttribute('href', 'feed:' + href);
        }
      });
  }
}
// Run when DOM is fully loaded
document.addEventListener('DOMContentLoaded', modifyFeedLinks);
// Also run when dynamic content is loaded (if using AJAX)
document.addEventListener('DOMNodeInserted', function(event) {
  if (event.target.getElementsByTagName) {
      modifyFeedLinks();
  }
});
`

The problem with this script the removal of base url from the url and replace it with feed:

**Reply by razonyang** [reply]
Does this URL (https://deploy-preview-1093--hbs-docs.netlify.app/) work?

**Reply by iamgiannisx** [reply]
Yes it's working.

**Reply by razonyang** [reply]
OK, should be fixed since 1.9.1

**Reply by iamgiannisx** [reply]
Thank you so much @razonyang
